neo/Consensus/ConsensusContext.cs

```diff
@@ -213,39 +213,35 @@ private void SignPayload(ConsensusPayload payload)
         internal void EnsureMaxBlockSize(IEnumerable<Transaction> txs)
         {
             var transactions = txs.ToList();
-
-            Transactions = new Dictionary<UInt256, Transaction>();
-            uint maxBlockSize = NativeContract.Policy.GetMaxBlockSize(Snapshot);
+            // Limit Speaker proposal to the limit `MaxTransactionsPerBlock` or all available transactions of the mempool
             TransactionHashes = new UInt256[Math.Min(transactions.Count, NativeContract.Policy.GetMaxTransactionsPerBlock(Snapshot))];
+            
             Block.Transactions = new Transaction[0];
-
             // TODO: Real expected witness
             Block.Witness = new Witness()
             {
                 InvocationScript = new byte[0],
                 VerificationScript = new byte[0]
             };
 
+            Transactions = new Dictionary<UInt256, Transaction>();
```

Comment:
@shargon, is this dictionary really needed?
It is being using during the resize, right?

Reply:
Is where is stored the transactions, is required, and acording to the count, the hashes will be truncated, is faster than a list

Reply:
I did not understand still, we only use TransactionHashes, @shargon.
Why not just truncate it?

Reply:
Is the same logic as before

https://github.com/neo-project/neo/pull/953/files#diff-b09366dde7105170c26fd3757e256035L218

Reply:
@shargon, it looks like to me that this var was obsolete before...
https://github.com/neo-project/neo/blob/master/neo/Consensus/ConsensusContext.cs
AHeuaheauhea

@erikzhang

Reply:
It was just initialized and set here:
https://github.com/neo-project/neo/blob/46ed67f6d4b551b2ef55c5daa0cfc8aefeea1981/neo/Consensus/ConsensusContext.cs#L30

https://github.com/neo-project/neo/blob/46ed67f6d4b551b2ef55c5daa0cfc8aefeea1981/neo/Consensus/ConsensusContext.cs#L216

very strange. But we can clean that in another PR, @shargon

Reply:
I see, Shargon, it is used when creating the block.

Reply:
The truncate using it confused me.
Can we change that for a static variable with the current size?